algorithms/unit_tests/TestStdAlgorithmsTeamInclusiveScan.cpp

```diff
@@ -0,0 +1,275 @@
+  // create a view in the memory space associated with default exespace
+  // with as many rows as the number of teams and fill it with random
+  // values from an arbitrary range.
+  constexpr ValueType lowerBound = 5;
+  constexpr ValueType upperBound = 523;
+  const auto bounds              = make_bounds(lowerBound, upperBound);
+
+  auto [sourceView, sourceViewBeforeOp_h] = create_random_view_and_host_clone(
+      LayoutTag{}, numTeams, numCols, bounds, "sourceView");
+
+  // -----------------------------------------------
+  // launch kokkos kernel
+  // -----------------------------------------------
+  using space_t = Kokkos::DefaultExecutionSpace;
+  Kokkos::TeamPolicy<space_t> policy(numTeams, Kokkos::AUTO());
+
+  // create the destination view
+  Kokkos::View<ValueType**> destView("destView", numTeams, numCols);
+
+  // inclusive_scan returns an iterator so to verify that it is correct
+  // each team stores the distance of the returned iterator from the beginning
+  // of the interval that team operates on and then we check that these
+  // distances match the std result
+  Kokkos::View<std::size_t*> distancesView("distancesView", numTeams);
+  // sentinel to check if all members of the team compute the same result
+  Kokkos::View<bool*> intraTeamSentinelView("intraTeamSameResult", numTeams);
+
+  PlusFunctor<ValueType> binaryOp;
+
+  // Create view of reduce init values to be used by test cases
+  Kokkos::View<ValueType*, Kokkos::DefaultHostExecutionSpace> initValuesView_h(
+      "initValuesView_h", numTeams);
+  using rand_pool =
+      Kokkos::Random_XorShift64_Pool<Kokkos::DefaultHostExecutionSpace>;
+  rand_pool pool(lowerBound * upperBound);
+  Kokkos::fill_random(initValuesView_h, pool, lowerBound, upperBound);
+
+  // use CTAD for functor
+  auto initValuesView =
+      Kokkos::create_mirror_view_and_copy(space_t(), initValuesView_h);
+  TestFunctorA fnc(sourceView, destView, distancesView, intraTeamSentinelView,
+                   initValuesView, binaryOp, apiId);
+  Kokkos::parallel_for(policy, fnc);
+
+  // -----------------------------------------------
+  // run cpp-std kernel and check
+  // -----------------------------------------------
+  auto distancesView_h         = create_host_space_copy(distancesView);
+  auto intraTeamSentinelView_h = create_host_space_copy(intraTeamSentinelView);
+  Kokkos::View<ValueType**, Kokkos::HostSpace> stdDestView("stdDestView",
+                                                           numTeams, numCols);
+
+  for (std::size_t i = 0; i < sourceView.extent(0); ++i) {
+    auto srcRow    = Kokkos::subview(sourceViewBeforeOp_h, i, Kokkos::ALL());
+    auto first     = KE::begin(srcRow);
+    auto last      = KE::end(srcRow);
+    auto destRow   = Kokkos::subview(stdDestView, i, Kokkos::ALL());
+    auto firstDest = KE::begin(destRow);
+    auto initValue = initValuesView_h(i);
+
+    ASSERT_TRUE(intraTeamSentinelView_h(i));
+
+#if defined(__GNUC__) && __GNUC__ == 8
```

Comment:
Again, check for the standard library version, not the compiler.

Reply:
Yes, leftover from before , will fix it thank you for pointing it out!

Reply:
fixed